src/cpufreq.c

```diff
@@ -168,27 +170,27 @@ static int cpufreq_read(void) {
          * we have to divide by 100. To get percents we have to multiply it
          * by 100 back. So, just use parsed value directly.
          */
-        if (!sscanf(buffer, "%s%llu", state, &time)) {
+        if (!sscanf(buffer, "%u%llu", &frequency, &time)) {
           ERROR("cpufreq plugin: Reading \"%s\" failed.", filename);
           break;
         }
 
-        if (j < MAX_AVAIL_FREQS) {
-          gauge_t g;
-          if (value_to_rate(&g, (value_t){.counter = time}, DS_TYPE_COUNTER,
-                            now, &(cpu_data[i].time_state[j])) != 0) {
-            j++;
-            continue;
-          }
-          cpufreq_submit(i, "percent", state, &(value_t){.gauge = g});
-        } else {
+        snprintf(state, sizeof(state), "%u", frequency);
```

Comment:
Please declare `state` just before this line.

https://collectd.org/review-comments#define-variables-on-first-use

Reply:
Updated.